src/components/Editor/SymbolModal.js

```diff
@@ -0,0 +1,291 @@
+    self.updateResults = this.updateResults.bind(this);
+    self.traverseResults = this.traverseResults.bind(this);
+    self.renderResults = this.renderResults.bind(this);
+    self.buildSummaryMsg = this.buildSummaryMsg.bind(this);
+    self.buildPlaceHolder = this.buildPlaceHolder.bind(this);
+  }
+
+  componentWillUnmount() {
+    const shortcuts = this.context.shortcuts;
+    shortcuts.off("Escape");
+    shortcuts.off(L10N.getStr("symbolSearch.search.key2"));
+  }
+
+  componentDidMount() {
+    const shortcuts = this.context.shortcuts;
+    shortcuts.on("Escape", (_, e) => this.onEscape(e));
+    shortcuts.on(L10N.getStr("symbolSearch.search.key2"), (_, e) => {
+      this.props.toggleActiveSearch("symbol");
+      this.props.setSelectedSymbolType("functions");
+    });
+  }
+
+  componentDidUpdate(prevProps: any, prevState: any) {
+    if (this.refs.resultList && this.refs.resultList.refs) {
+      scrollList(this.refs.resultList.refs, this.state.selectedResultIndex);
+    }
+  }
+
+  onEscape(e: SyntheticEvent) {
+    e.stopPropagation();
+    e.preventDefault();
+    this.props.toggleActiveSearch();
+  }
+
+  onChange(e: SyntheticEvent) {
+    const { selectedSource, setSymbolSearchQuery } = this.props;
+    if (!selectedSource || !selectedSource.get("text")) {
+      return;
+    }
+
+    setSymbolSearchQuery(e.target.value);
+
+    return this.updateResults(e.target.value);
+  }
+
+  updateResults(query: string) {
+    const {
+      selectedSource,
+      updateSearchResults,
+      updateSymbolSearchResults,
+      selectedSymbolType,
+      symbols
+    } = this.props;
+
+    console.log(query, selectedSource);
+
+    if (query == "" || !selectedSource) {
+      return;
+    }
+
+    const symbolSearchResults = filter(symbols[selectedSymbolType], query, {
+      key: "value"
+    });
+
+    updateSearchResults({ count: symbolSearchResults.length });
+    updateSymbolSearchResults(symbolSearchResults);
+  }
+
+  traverseResults(rev: boolean) {
+    const { symbolSearchResults, selectedResultIndex } = this.state;
+    const searchResults = symbolSearchResults;
+    const resultCount = searchResults.length;
+
+    if (rev) {
+      let nextResultIndex = Math.max(0, selectedResultIndex - 1);
+
+      if (selectedResultIndex === 0) {
+        nextResultIndex = resultCount - 1;
+      }
+      this.setState({ selectedResultIndex: nextResultIndex });
+      this.onSelectResultItem(searchResults[nextResultIndex]);
+    } else {
+      let nextResultIndex = Math.min(resultCount - 1, selectedResultIndex + 1);
+
+      if (selectedResultIndex === resultCount - 1) {
+        nextResultIndex = 0;
+      }
+      this.setState({ selectedResultIndex: nextResultIndex });
+      this.onSelectResultItem(searchResults[nextResultIndex]);
+    }
+  }
+
+  onKeyDown(e: SyntheticKeyboardEvent) {
+    const { enabled, query } = this.props;
+    const { symbolSearchResults } = this.state;
+    if (!enabled || query == "") {
```

Comment:
we can use identity equals here `===` since `"" == 0 // true`

or if we don't care about type conversion we can just `(!enabled || !query)`